### PR TITLE
Add support for trace counters with variable arguments and instrument the raster cache.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: fml/trace_event.cc [hunk 1]
```diff
@@ -4,23 +4,36 @@
...
+void TraceTimelineEvent(TraceArg category_group,
+                        TraceArg name,
+                        TraceIDArg identifier,
+                        Dart_Timeline_Event_Type type,
+                        const std::vector<const char*>& names,
+                        const std::vector<std::string>& values) {
+  const auto argument_count = std::min(names.size(), values.size());
+
+  auto c_values =
+      static_cast<const char**>(calloc(argument_count, sizeof(const char*)));
```
Comment (jason-simmons) [inline fml/trace_event.cc]: Can you use a std::vector<const char*> here?

Comment (chinmaygarde) [reply fml/trace_event.cc]: Yeah.

Comment (chinmaygarde) [reply fml/trace_event.cc]: Done

**Comment by chinmaygarde** [comment]
@rmacnak-google explained to me earlier that the keys are not actually copied by the timeline. I have not found a way to enforce this requirement via a static assert yet however.

**Comment by chinmaygarde** [comment]
xref https://dart-review.googlesource.com/c/sdk/+/96145

**Pull request merged**